backend/src/Designer/EventHandlers/IFileSyncHandlerExecutor.cs

```diff
@@ -8,4 +8,8 @@ public interface IFileSyncHandlerExecutor
 {
     Task ExecuteWithExceptionHandling(AltinnRepoEditingContext editingContext, string errorCode, string sourcePath,
         Func<Task> handlerFunction);
+
+    Task ExecuteWithExceptionHandlingConditionalNotification(AltinnRepoEditingContext editingContext, string errorCode, string sourcePath,
```

Comment:
It's unclear from the definition that the handlerFunction function is responsible for deciding if it should notify clients. Can you add docs/summary and describe parameters?
Can old method be deleted? This seems like a correct way to do it. Notify only if there is an actual change in the file.